include/shell/shell.h

```diff
@@ -346,6 +345,15 @@ struct shell_ctx {
 
 extern const struct log_backend_api log_backend_shell_api;
 
+/**
+ * @brief Flags for setting shell output newline sequence.
+ */
+enum shell_flag {
+	SHELL_FLAG_LFCR_DEFAULT	= (1<<0),	/* Do not map LF or CR */
+	SHELL_FLAG_OLFCR	= (1<<1),	/* Map LF to LFCR on output */
```

Comment:
Should this be ``Map LF to CRLF on output``

Reply:
@dbkinder 
In the website you have linked (wikipedia) there is also a valid sequence `\n\r`. 
I've tested it on serial terminal. CR brings cursor to the beginning of the line, LF moves cursor to the new line.
CR and LF are only instructions for the terminal to drive a cursor, order should not matter.

Some "inteligent" terminals when see CR adds silently LF while others when see LF adds CR.

Reply:
I can change it, this will require slightly code modification. Instead of checking for the previous character I will need to verify successor. However to be honest I do not see any case where current implementation will not work correctly.

Reply:
Order does matter when applications are expecting either a LF (``\n``) or a CRLF (``\r\n``) line termination. That said, I'll abide by the decision of the developer community on this.  :)

Reply:
@carlescufi @dbkinder : I fixed according to your proposals.
To not mix things up there are supported 2 modes:

- SHELL_FLAG_LFCR_DEFAULT- no characters added by shell
- SHELL_FLAG_OLF_CRLF - shell is parsing an output string and before each `\n` it will add `\r`